src/themes/themesController.ts

```diff
@@ -0,0 +1,341 @@
+
+        let newHyperLinkStyle = `.${ styles.hyperLinkTheme } a {
+            color: ${ theme["hyperlink-font-color"] } !important;
+        }`;
+        newStyles += newHyperLinkStyle;
+        
+        newStyles += this.setDialogStyle(theme);
+        newStyles += this.setPrimaryBtnStyle(theme);
+        newStyles += this.setSecondaryBtnStyle(theme);
+        newStyles += this.setRadioBtnStyle(theme);
+
+        document.getElementById('ms-consent-banner-theme-styles')!.innerHTML = newStyles;
+    }
+
+    /**
+     * Set missing color from another provided color
+     * 
+     * e.g.: 
+     * provider: #ffffff; missing: rgba(255, 255, 255, transparencyFactor)
+     * provider: rgb(255, 255, 255); missing: rgba(255, 255, 255, transparencyFactor)
+     * provider: rgb(255, 255, 255, 1); missing: rgba(255, 255, 255, transparencyFactor)
+     * 
+     * @param {keyof ITheme} missing missing property in the theme
+     * @param {string} provider the provided property in the theme
+     * @param {number} transparencyFactor the alpha channel that will be used in the missing property
+     * @param {ITheme} currentTheme the target theme that will be set
+     */
+    private static setMissingColorFromAnotherProperty(missing: keyof ITheme,
+                                               provider: string,
+                                               transparencyFactor: number,
+                                               currentTheme: ITheme): void {
+
+        // provider: #ffffff; missing: rgba(255, 255, 255, transparencyFactor)
+        if (provider.startsWith("#")) {
+            let hexColor = parseInt(provider.substring(1), 16);
+            let r = (hexColor >> 16) & 255;
+            let g = (hexColor >> 8) & 255;
+            let b = hexColor & 255;
+
+            currentTheme[missing] = "rgba(" + r + ", " + g + ", " + b + ", " + transparencyFactor + ")";
+        }
+        // provider: rgb(255, 255, 255); missing: rgba(255, 255, 255, transparencyFactor)
+        else if (provider.startsWith("rgb(")) {
+            let missingColor = "rgba" + provider.substring(3, provider.length - 1) + ", " + transparencyFactor + ")";
+            currentTheme[missing] = missingColor;
+        }
+        // provider: rgb(255, 255, 255, 1); missing: rgba(255, 255, 255, transparencyFactor)
+        else if (provider.startsWith("rgba")) {
+            let rgbIdx = provider.lastIndexOf(",");
+            let transparency = provider.substring(rgbIdx + 1, provider.length - 1);
+            
+            let newTransparency = parseFloat(transparency.trim()) * transparencyFactor;
+            let missingColor = provider.substring(0, rgbIdx + 1) + newTransparency.toFixed(2) + ")";
+            currentTheme[missing] = missingColor;
+        }
+    }
+
+    /**
+     * Apply the theme to dialog
+     * 
+     * @param {ITheme} theme theme that will be applied to dialog
+     */
+    private static setDialogStyle(theme: ITheme): string {
```

Comment:
Plz rename all `set...` functions to `build...` since they're building the styles string, and not setting anything.